src/targets/services/categorize.js

```diff
@@ -5,39 +5,59 @@ import CozyClient, { Q } from 'cozy-client'
 import { BANK_DOCTYPE } from '../../doctypes'
 import { Model } from './helpers'
 import dissecConfig from '../../../dissec.config.json'
+import { JOBS_DOCTYPE } from '../../doctypes/jobs'
 
 export const categorize = async () => {
-  const { pretrained } = JSON.parse(process.env['COZY_PAYLOAD'] || '{}')
-
   const client = CozyClient.fromEnv(process.env, {})
 
-  // 1. Fetch data
+  // Fetching parameters (if any) from the jobs
+  const { data: job } = await client.query(
+    Q(JOBS_DOCTYPE).getById(process.env['COZY_JOB_ID'].split('/')[2])
+  )
+
+  const { pretrained, filters = {} } = job.attributes.message
+
+  // Fetch data
   const { data: operations } = await client.query(Q(BANK_DOCTYPE))
 
-  // 2. Fetch model or initialize it
+  // Fetch model or initialize it
   let model
   if (pretrained) {
-    // Use the stack's remote assets
+    // Use the shared model
     try {
-      const compressedBackup = fs
+      const compressedAggregate = fs
         .readFileSync(dissecConfig.localModelPath)
         .toString()
-      model = Model.fromCompressedBackup(compressedBackup)
+      model = Model.fromCompressedAggregate(compressedAggregate)
     } catch (err) {
-      throw `Model does not exist at path ${dissecConfig.localModelPath}`
+      throw `Model does not exist at path ${
+        dissecConfig.localModelPath
+      } ? ${err}`
     }
   } else {
-    model = Model.fromDocs(operations)
+    // Apply filters first
+    let filteredOperations = operations
+    if (filters.date) {
+      filteredOperations = filteredOperations.filter(
+        e =>
+          new Date(filters.date).valueOf() -
+            new Date(e.cozyMetadata.createdAt).valueOf() <=
+          0
+      )
+    }
```

Comment:
Mango query to the rescue : https://docs.cozy.io/en/tutorials/data/queries/
You can directly query by date and avoid the hassle to do it on the client side ; it is also much more efficient, this is the job of the database after all ;)
```
const query = Q(BANK_DOCTYPE).where({date: $gt: { filters.minOperationDate }}).sortBy([{date: "asc"}])
const { data: operations } = await client.query(query)
```